content_restrictions/content_restrictions.py

```diff
@@ -168,11 +215,66 @@ def has_access_to_content(self):
         Check if the user has access to the content.
 
         - If password restriction is enabled, check if the user has entered the correct password.
+        - If IP restriction is enabled, check if the user has access with the IP whitelist.
         """
-        if self.password_restriction:
-            return bool(self.user_provided_password == self.password)
+        if self.password_restriction and not self.is_correct_password:
+            return False
+        if self.ip_restriction:
+            return self.has_access_with_ip_whitelist(get_current_request())
         return True
 
+    @property
+    def is_correct_password(self) -> bool:
+        """
+        Check if the user has entered the correct password.
+
+        Returns:
+            bool: True if the user has entered the correct password, False otherwise.
+        """
+        return self.user_provided_password == self.password
+
+    def has_access_with_ip_whitelist(self, request) -> bool:
+        """
+        Check if the user has access to the content based on the IP whitelist.
+
+        Args:
+            request (WSGIRequest): The current request object.
+
+        Returns:
+            bool: True if the user has access to the content, False otherwise.
+        """
+        client_ips = ip.get_all_client_ips(request)
+        for ip_add_or_range in self.ip_whitelist:
+            if any(self.ip_has_access(client_ip, ip_add_or_range) for client_ip in client_ips):
+                return True
+        return False
+
+    @staticmethod
+    def ip_has_access(client_ip: str, ip_address_or_range: str) -> bool:
+        """
+        Check if the client IP matches the IP address or is in the IP range.
+
+        The IP address or range can be in IPv4 or IPv6 format.
+
+        Args:
+            client_ip (str): The IP address to check.
+            ip_address_or_range (str): The IP address or IP range to check against.
+
+        Returns:
+            bool: True if the IP is in the network, False otherwise.
+        """
+        client_ip = ipaddress.ip_address(client_ip)
+
+        try:
+            ip_address = ipaddress.ip_address(ip_address_or_range)
+            return client_ip == ip_address
+        except ValueError:
+            try:
+                ip_network = ipaddress.ip_network(ip_address_or_range, strict=False)
+                return client_ip in ip_network
+            except ValueError:
+                return False
```

Comment:
Can we log this exception? So it's known there's something wrong with the configurations.

Reply:
Done: https://github.com/eduNEXT/xblock-content-restrictions/commit/f23e1c7161fe94b83da43ecfbd528b638053df3c